### ISSUE TITLE
nixos-option fails if configuration has trace output

### ISSUE THREAD

**Issue by deliciouslytyped** [NixOS/nixpkgs#67659]
Repro script (run with `nix-shell -A success` or `-A fail`):
```nix
let
#pinning
  pkgs' = import <nixpkgs> {};
  pkgs = import (pkgs'.fetchFromGitHub {
    owner = "NixOS";
    repo = "nixpkgs";
    rev = "33b67761be99c21b151b9a3b8c5c8bd6398671a9";
    sha256 = "1i3yxywwivcj4ls7pyvy696m44g9xww7khx71vpm2igj4i9b7xq1";
    }) {};

#get command
  nixos-option = pkgs.substituteAll {
    name = "nixos-option.sh";
    isExecutable = true;
    src = <nixpkgs/nixos/modules/installer/tools/nixos-option.sh>;
    };

#repro
  configuration = fail: pkgs.writeText "configuration.nix" ''
    {pkgs, lib, ...}: let fail = ${if fail then "true" else "false"}; in {
      #the trace needs to come from somewhere that's not the examined attribute for some reason?
      imports = [ ({...}: if fail then (lib.traceVal {}) else {}) ]; 
      boot.loader.grub.devices = [ "nodev" ];
      fileSystems."/" = {};
      }
    '';
  test = fail: ''
    ${nixos-option} -I nixos-config=${configuration fail} boot.loader.grub.devices --verbose
    exit
    '';
in {
  fail = pkgs.mkShell {
    shellHook = test true;
    };
  success = pkgs.mkShell {
    shellHook = test false;
    };  
  }


```

The solution is probably to do something saner than merge all the output streams in `evalNix`?
The error seems to originate via the trace line getting substituted into the end of the header in `generateAccessors`, resulting in things like 
```bash
header='let
  nixos = import <nixpkgs/nixos> {};
  nixpkgs = import <nixpkgs> {};
in with nixpkgs.lib;

trace: { } let option = nixos.options."boot"."loader"."grub"."devices"; config = nixos.config."boot"."loader"."grub"."devices"; in
'
```

**Comment by deliciouslytyped** [comment]
@nbp because git blame.

**Comment by Ma27** [comment]
IIRC traces go to `stderr`, while the actual output goes to `stdout`. As I hacked around in `nixos-option` as well, I can probably take a look at this during the next days...

**Comment by Ma27** [comment]
@deliciouslytyped would you mind having a look at #68121

**Comment by bjornfor** [comment]
This should be fixed on master and release-19.09 with a3961978713624d9d218f0699cbe8007937a296f. Closing.

**Comment by deliciouslytyped** [comment]
For my later reference https://github.com/NixOS/nixpkgs/pull/68193
